### PR TITLE
Serving/batch serving feature spark

### PR DESCRIPTION
**Pull request by RiddhiBagadiaa**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

**Review by ahmadnazeri — APPROVED** [review]

File: provider/spark.go [hunk 1]
```diff
@@ -669,8 +668,110 @@ func (store *SparkOfflineStore) AsOfflineStore() (OfflineStore, error) {
...
+		materializationIDs[i] = ResourceID{Name: id.Name, Variant: id.Variant, Type: FeatureMaterialization}
+		batchDir += fmt.Sprintf("%s-%s", id.Name, id.Variant)
+		if i != len(ids)-1 {
+			batchDir += "-"
+		}
+	}
+	// Convert materialization ID to file paths
+	materializationPaths := make([]string, len(ids))
+	for i, id := range materializationIDs {
+		path, err := store.Store.CreateDirPath(id.ToFilestorePath())
```
Comment (sdreyer) [inline provider/spark.go:688]: Make everything in this loop into a function

Comment (sdreyer) [reply provider/spark.go:688]: Should also check that they all have the same offline store

File: serving/serving.go [hunk 2]
```diff
@@ -172,6 +172,26 @@ func (serv *FeatureServer) getBatchFeatureIterator(ids []provider.ResourceID) (p
 	if err != nil {
 		return nil, errors.Wrap(err, "could not fetch provider")
 	}
+  
+	providerName := providerEntry.Name()
+
+	for id := range ids {
```
Comment (sdreyer) [inline serving/serving.go:178]: Make this into a function called `checkFeatureSources(ids)`

File: provider/spark.go [hunk 3]
```diff
@@ -669,8 +668,110 @@ func (store *SparkOfflineStore) AsOfflineStore() (OfflineStore, error) {
...
+	}
+	iterator, err := store.Store.Serve(outputFiles)
+	if err != nil {
+		return nil, fmt.Errorf("could not serve batch features: %w", err)
+	}
+	store.Logger.Debug("Successfully created batch iterator")
+	return &FileStoreBatchServing{store: store.Store, iter: iterator, numFeatures: len(ids)}, nil
+}
+
+func createJoinQuery(numFeatures int) string {
```
Comment (aolfat) [inline provider/spark.go:745]: nit: might want to use a stringbuilder instead

**Pull request closed without merging**